src/app/modules/submission/encrypt-submission/encrypt-submission.utils.ts

```diff
@@ -210,3 +219,104 @@ export const createEncryptedSubmissionDto = (
     attachmentMetadata: attachmentPresignedUrls,
   }
 }
+
+export class IncomingEncryptSubmission extends IncomingSubmission {
+  constructor(
+    responses: FieldResponse[],
+    public readonly form: IPopulatedEncryptedForm,
+    public readonly encryptedContent: string,
+  ) {
+    super(responses, form)
+  }
+
+  static init(
+    form: IPopulatedEncryptedForm,
+    responses: FieldResponse[],
+    encryptedContent: string,
+  ): Result<
+    IncomingEncryptSubmission,
+    ProcessingError | ConflictError | ValidateFieldError
+  > {
+    const logMeta = {
+      action: 'IncomingEncryptSubmission initialisation',
+      formId: form._id,
+    }
+    return checkIsEncryptedEncoding(encryptedContent)
+      .mapErr((error) => {
+        logger.error({
+          message: 'Error verifying content has encrypted encoding.',
+          meta: logMeta,
+          error,
+        })
+        return error
+      })
+      .andThen(() =>
+        this.processResponses(responses, form).mapErr((error) => {
+          logger.error({
+            message: 'Error processing encrypted submission.',
+            meta: logMeta,
+            error,
+          })
+          return error
+        }),
+      )
+      .map(
+        (filteredResponses) =>
+          new IncomingEncryptSubmission(
+            filteredResponses,
+            form,
+            encryptedContent,
+          ),
+      )
+  }
+
+  private static processResponses(
+    responses: FieldResponse[],
+    form: IPopulatedEncryptedForm,
+  ): Result<
+    FieldResponse[],
+    ProcessingError | ConflictError | ValidateFieldError
+  > {
+    return getFilteredResponses(form, responses)
+      .andThen((filteredResponses) =>
+        this.getFieldMap(form, filteredResponses).map((fieldMap) => ({
+          filteredResponses,
+          fieldMap,
+        })),
+      )
+      .map(({ filteredResponses, fieldMap }) => ({
+        filteredResponses,
+        fieldMap,
+        visibleFieldIds: getVisibleFieldIds(filteredResponses, form),
+        verifiableResponseIds: this.getVerifiableResponseIds(
+          filteredResponses,
+          fieldMap,
+        ),
+        visibleResponseIds: this.getVisibleResponseIds(
+          filteredResponses,
+          fieldMap,
+          (response: FieldResponse) =>
+            'answer' in response &&
+            typeof response.answer === 'string' &&
+            response.answer.trim() !== '',
+        ),
+      }))
```

Comment:
suggest creating this block into a smaller readable function like
```ts
const transformFilteredResponses = ({filteredResponses, form, fieldMap}) => { ... }
```

then the code would have a better visual flow:

```ts
.andThen((filteredResponses) =>
  this.getFieldMap(form, filteredResponses).map((fieldMap) => 
    transformFilteredResponses({filteredResponses, fieldMap, form})
)
```